table.cc

```diff
@@ -2476,15 +2476,29 @@ table::make_reader_excluding_sstable(schema_ptr s,
     return make_combined_reader(s, std::move(readers), fwd, fwd_mr);
 }
 
-void table::move_sstable_from_staging_in_thread(sstables::shared_sstable sst) {
-    try {
-        sst->move_to_new_dir_in_thread(dir(), sst->generation());
-    } catch (...) {
-        tlogger.warn("Failed to move sstable {} from staging: {}", sst->get_filename(), std::current_exception());
-        return;
-    }
-    _sstables_staging.erase(sst->generation());
-    _compaction_strategy.get_backlog_tracker().add_sstable(sst);
+future<> table::move_sstables_from_staging(std::vector<sstables::shared_sstable> sstables) {
+    return with_semaphore(_sstable_deletion_sem, 1, [this, sstables = std::move(sstables)] {
+        return do_with(std::set<sstring>({dir()}), [this, sstables = std::move(sstables)] (auto& dirs_to_sync) {
```

Comment:
nitpick: `auto&` parameters like this tend to confuse my IDE, could you replace it with `std::set<sstring>&`?